step-capstone/src/__tests__/SearchForEmptyTimeslot.test.js

```diff
@@ -0,0 +1,666 @@
+import "@testing-library/jest-dom";
```

Comment:
Please checkout this unit test file as well :D

Reply:
The test looks fine overall but there are some duplicate cases that looks like they could be eliminated. I see that there are a few foundational cases (clicking in between, clicking in, clicking before, clicking after an event, handling overflow events, etc) but it looks like the tests cover these for a different N number of events (2-5 events). If the cases are handled the same way regardless of N then it's probably better to have less tests for the same coverage.

Reply:
Hi Chris, thank you for your comments! I was just thinking it'll be better to test the binary search algorithm with different array length as well as different search positions. (array in this case is displayItems) Can you tell me about your thoughts? @cthamrun